doc/sphinx/addendum/type-classes.rst

```diff
@@ -15,34 +15,37 @@ Class and Instance declarations
 The syntax for class and instance declarations is the same as the record
 syntax of Coq:
 
-``Class Id (`` |p_1| ``:`` |t_1| ``) ⋯ (`` |p_n| ``:`` |t_n| ``) [:
-sort] := {`` |f_1| ``:`` |u_1| ``; ⋮`` |f_m| ``:`` |u_m| ``}.``
+.. coqdoc::
 
-``Instance ident : Id`` |p_1| ``⋯`` |p_n| ``:= {`` |f_1| ``:=`` |t_1| ``; ⋮`` |f_m| ``:=`` |t_m| ``}.``
+  Class classname (p1 : t1) ⋯ (pn : tn) [: sort] := { f1 : u1 ; ⋯ ; fm : um }.
 
-The |p_i| ``:`` |t_i| variables are called the *parameters* of the class and
-the |f_i| ``:`` |t_i| are called the *methods*. Each class definition gives
+  Instance instancename q1 ⋯ qm : classname p1 ⋯ pn := { f1 := t1 ; ⋯ ; fm := tm }.
```

Comment:
While this is already an improvement, the good way of documenting this would be:

```rst
.. cmd:: Class @ident @binders {? @sort } := { {*; @ident : @type } }

   This defines a new typeclass :token:`ident`.

.. cmd:: Instance @ident @binders : @qualid {* @term_i } := { {*; @qualid__j := @term_j } }

   This defines an instance :token:`ident` for class :n:`@qualid {* @term_i }`...
```

However, I realize that this is already present in the later "Summary of commands" section. Personally, I would recommend removing this section and documenting the syntax of the commands and variants earlier, as they are introduced. But this is a larger refactoring that you might not be prepared to tackle.